crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -367,6 +381,15 @@ impl SettlementEncoder {
                     .context("buy_amount computation failed")?
                     .checked_div(uniform_buy_price)
                     .context("buy_amount computation failed")?;
+
+                // Reduce the `buy_amount` by protocol fee
+                let surplus = buy_amount
+                    .checked_sub(order.data.buy_amount)
+                    .unwrap_or(0.into());
```

Comment:
For `sell` orders, realized `buy_amount` should be higher than limit price (`order.data.buy_amount`)